ocular/src/keyring.rs

```diff
@@ -0,0 +1,752 @@
+        }
+    }
+
+    fn rename_key(
+        &self,
+        current_name: &str,
+        new_name: &str,
+        override_if_exists: bool,
+    ) -> Result<(), KeyStoreError> {
+        // Check if current key exists
+        if !self.key_exists(current_name)? {
+            eprintln!("Key '{}', does not exist.", current_name);
+            return Err(KeyStoreError::DoesNotExist(current_name.to_string()));
+        }
+
+        // Check if new key exists
+        if self.key_exists(new_name)? && !override_if_exists {
+            eprintln!("New key name '{}', already exists.", new_name);
+            return Err(KeyStoreError::Exists(new_name.to_string()));
+        }
+
+        // Proceed with rename
+        let current_name = &KeyName::new(current_name)
+            .unwrap_or_else(|_| panic!("Could not create KeyName for '{}'.", current_name));
+        let new_name = &KeyName::new(new_name)
+            .unwrap_or_else(|_| panic!("Could not create KeyName for '{}'.", new_name));
+
+        let key = self
+            .key_store
+            .as_ref()
+            .expect("Error accessing key store.")
+            .load(current_name)
+            .expect("Could not load key.");
+
+        // Create new key.
+        let _result = self
+            .key_store
+            .as_ref()
+            .expect("Error accessing key store.")
+            .store(new_name, &key);
+
+        // Delete old key.
+        let _result = self.delete_key(current_name);
+
+        Ok(())
+    }
+
+    fn get_key_address(&self, name: &str) -> Result<PublicKeyOutput, KeyStoreError> {
+        // Check if key exists
+        if !self.key_exists(name)? {
+            eprintln!("Key '{}', does not exist.", name);
+            return Err(KeyStoreError::DoesNotExist(name.to_string()));
+        }
+
+        // Prepare key and parse into pem
+        let key_name = &KeyName::new(name)
+            .unwrap_or_else(|_| panic!("Could not create KeyName for '{}'.", name));
+        let key = self
+            .key_store
+            .as_ref()
+            .expect("Error accessing key store.")
+            .load(key_name)
+            .expect("Could not load key");
+        let key: k256::elliptic_curve::SecretKey<k256::Secp256k1> = key
+            .to_pem(LineEnding::default())
+            .expect("Could not convert to pem.")
+            .parse()
+            .expect("Could not parse pem");
+
+        // Convert to deepspace private key for address conversion
+        let key_bytes = key.to_bytes();
+
+        // Obtain public key bytes
+        let secp256k1_engine = secp256k1::Secp256k1::new();
+        let secp256k1_secret_key =
+            secp256k1::SecretKey::from_slice(&key_bytes).expect("Could not create secret key.");
+        let secp256k1_public_key =
+            secp256k1::PublicKey::from_secret_key(&secp256k1_engine, &secp256k1_secret_key);
+        let secp256k1_public_key_bytes = secp256k1_public_key.serialize();
```

Comment:
I think you want:

```suggestion
    let verifying_key = key.verifying_key();
```

Also I'd suggest naming `key` something like `signing_key` instead to keep it unambiguous.

Reply:
Changed this up to use cosmrs and renamed the vars to have more consistent and clear naming conventions.